### PR TITLE
[starlink] Veggiesaurus/starlink ast cminpack fix

### PR DESCRIPTION
**Pull request by veggiesaurus**
Minor fix for undefined/missing minpack functions when linking AST.

- #### What does your PR fix?  
Package now depends on `cminpack` and uses the `--with-external-cminpack` configure flag (added to AST 9.2.5 for this PR 😄)

- #### Which triplets are supported/not supported?
No change

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
Yes

### CODE REVIEW

**Review by NancyLi1013 — APPROVED** [review]

File: ports/starlink-ast/vcpkg.json [hunk 1]
```diff
@@ -1,10 +1,13 @@
 {
+  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg/master/scripts/vcpkg.schema.json",
```
Comment (JackBoosY) [inline ports/starlink-ast/vcpkg.json:2]:
```suggestion
```
We don't need this comment.

**Review by JackBoosY — CHANGES_REQUESTED** [review]
One minor thing last.

File: ports/starlink-ast/portfile.cmake [hunk 2]
```diff
@@ -1,9 +1,9 @@
 vcpkg_fail_port_install(ON_TARGET "Linux" "OSX")
 
 vcpkg_download_distfile(ARCHIVE
-    URLS "https://github.com/Starlink/ast/releases/download/v9.2.4/ast-9.2.4.tar.gz"
-    FILENAME "ast-9.2.4.tar.gz"
-    SHA512 84e6f243e6d9d77328b73b97355feb3990307fb9c8f9b2f30344d71e2f5e63a849cdce0090ff5b7cc16028e12d68516c885b13d76db841072c9d1d06a7742a9e
+    URLS "https://github.com/Starlink/ast/releases/download/v9.2.5/ast-9.2.5.tar.gz"
+    FILENAME "ast-9.2.5.tar.gz"
+    SHA512 2eee3df56ca49a9fbb6b103b26f41a63a7b77533bcc74bc4154b57567ee6da53d5c7ff1d38c4ad0ce330d9b902e35d40184fa9e985506c3dcbbb03071be65760
```
Comment (JackBoosY) [inline ports/starlink-ast/portfile.cmake:6]: Oh I think you should use the following code instead:
```cmake
vcpkg_from_github(
    OUT_SOURCE_PATH SOURCE_PATH
    REPO Starlink/ast
    REF  v9.2.5
    SHA512 2eee3df56ca49a9fbb6b103b26f41a63a7b77533bcc74bc4154b57567ee6da53d5c7ff1d38c4ad0ce330d9b902e35d40184fa9e985506c3dcbbb03071be65760
    HEAD_REF master
)
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/starlink-ast/portfile.cmake (hunk 2 above)
Comment (veggiesaurus) [reply ports/starlink-ast/portfile.cmake:6]: That results in a _different_ file being downloaded (with a hash of `1a0acffb0fed1ac848362b60897b6ef6a5fbbb3a4990a7f2abc4dbeae7fddb923ee27986481304acb2f67d93497c173a525d883d550e90243f3ccb3fcef50f09)`. After updating the hash, it downloads ok, but autoconf step fails with the following error:

```
configure.ac:27: error: require Automake 1.8.2-starlink, but have 1.16.3
```

I think we should rather stick to the official release file.

Comment (dg0yt) [reply ports/starlink-ast/portfile.cmake:6]: `vcpkg_from_github` downloads raw sources which need  `autoreconf` before `configure`. 
The release tarball contains the result of running `autoreconf`. It looks like the maintainer uses a very special version of `automake`.

Comment (veggiesaurus) [reply ports/starlink-ast/portfile.cmake:6]: There are a number of other requirements as far as I can tell. Hence the simplest solution is to just download the release tarball

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**